drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/ClasspathKieProject.java

```diff
@@ -440,41 +440,50 @@ public static String fixURLFromKProjectPath(URL url) {
     }
 
     private static String getPathForVFS(URL url) {
-        Method m = null;
+        Method vfsGetPhysicalFileMethod = null;
         try {
-            m = Class.forName("org.jboss.vfs.VirtualFile").getMethod("getPhysicalFile");
+            vfsGetPhysicalFileMethod = Class.forName("org.jboss.vfs.VirtualFile").getMethod("getPhysicalFile");
         } catch (Exception e) {
             try {
                 // Try to retrieve the VirtualFile class also on TCCL
-                m = Class.forName("org.jboss.vfs.VirtualFile", true, Thread.currentThread().getContextClassLoader()).getMethod("getPhysicalFile");
+                vfsGetPhysicalFileMethod = Class.forName("org.jboss.vfs.VirtualFile", true, Thread.currentThread().getContextClassLoader()).getMethod("getPhysicalFile");
             } catch (Exception e1) {
                 // VirtualFile is not available on the classpath - ignore
                 log.warn( "Found virtual file " + url + " but org.jboss.vfs.VirtualFile is not available on the classpath" );
             }
         }
-        Method m2 = null;
+        
+        Class vfsClass = null;
+        Method vfsGetChildMethod = null;
+        boolean useTccl = false;
```

Comment:
@tkobayas this is the only relevant difference with your solution: we need to know from which classloader we loaded the VFS classes and always use the same consistently. I believe this is necessary for OSGi.